theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
Adding to this.
@f0k has explained it quite well here https://github.com/Theano/Theano/issues/5621#issuecomment-282803917, here he switched the dimensions of inputs and outputs though